AnagramSolver.WebApp/Controllers/Api/AnagramApiController.cs

```diff
@@ -42,14 +57,47 @@ public string GetJsonString(string id)
         [HttpGet("[action]/{id}")]
         public List<string> GetForJS(string id)
         {
+            var stopWatch = new Stopwatch();
+            stopWatch.Start();
+            var numberOfAnagrams = 0;
             var listOfAnagrams = new List<string>();
             var allWords = _wordRepository.GetAllWords();
             var _anagramSolver = new BusinessLogic.Classes.AnagramSolver(allWords);
-            var anagramsById = _anagramSolver.GetAnagrams(id);
+            List<string> anagramsById;
+            if (_cachedAnagrams.GetCachedAnagram(id) != null)
+            {
+                anagramsById = _cachedAnagrams.GetCachedAnagram(id);
+            }
+            else
+            {
+                anagramsById = _anagramSolver.GetAnagrams(id);
+                _cachedAnagrams.PutAnagramToCache(id, anagramsById);
+            }
             foreach (var word in anagramsById)
             {
+                numberOfAnagrams++;
```

Comment:
Why each time `++` while you can just call `listOfAnagrams.Count()`?

Reply:
fixed